src/lib/components/search-modal.svelte

```
@@ -0,0 +1,217 @@
+        }
+    }
+
+    let pendingNavigation = false;
+
+    function handleResultClick(item: TMDBTransformedListItem) {
+        savedQuery = query;
+        savedResults = results;
+        navigatedFromModal = true;
+        pendingNavigation = true;
+        // Keep modal open — it will be hidden after the new page loads
+        goto(`/details/media/${item.id}/${item.media_type}`);
```

ESLint

[error] Unexpected goto() call without resolve().

(svelte/no-navigation-without-resolve)